src/Methods/EloquentBuilderForwardsCallsExtension.php

```diff
@@ -88,18 +89,24 @@ public function getMethod(ClassReflection $classReflection, string $methodName):
 
         $templateTypeMap = $classReflection->getActiveTemplateTypeMap();
 
-        /** @var Type|ObjectType|null $modelType */
+        /** @var Type|ObjectType|TemplateMixedType $modelType */
         $modelType = $templateTypeMap->getType('TModelClass');
 
-        if ($this->getBuilderReflection()->hasNativeMethod($methodName) && ($modelType === null || ! $modelType instanceof ObjectType)) {
+        if ($this->getBuilderReflection()->hasNativeMethod($methodName) && (! $modelType instanceof ObjectType)) {
             $methodReflection = $this->getBuilderReflection()->getNativeMethod($methodName);
+            $builderClass = EloquentBuilder::class;
+
+            if ($modelType instanceof TemplateMixedType) {
+                /** @var string $builderClass */
+                $builderClass = $modelType->getScope()->getClassName();
```

Comment:
Could class name be a non-string?

Reply:
Yes. Technically it can be `null`. But in reality, it'll never be. So this comment is just to tell PHPStan that it's a string.

Reply:
Thanks.